python/lsst/afw/cameraGeom/_detector.cc

```diff
@@ -0,0 +1,209 @@
+        cls.def("getPhysicalType", &DetectorBase::getPhysicalType);
+        cls.def("getSerial", &DetectorBase::getSerial);
+        cls.def("getBBox", &DetectorBase::getBBox);
+        cls.def("getOrientation", &DetectorBase::getOrientation);
+        cls.def("getPixelSize", &DetectorBase::getPixelSize);
+        cls.def("hasCrosstalk", &DetectorBase::hasCrosstalk);
+        cls.def("getCrosstalk", &DetectorBase::getCrosstalk);
+        cls.def("getNativeCoordSys", &DetectorBase::getNativeCoordSys);
+        cls.def("makeCameraSys",
+                py::overload_cast<CameraSys const &>(&DetectorBase::makeCameraSys, py::const_),
+                "cameraSys"_a);
+        cls.def("makeCameraSys",
+                py::overload_cast<CameraSysPrefix const &>(&DetectorBase::makeCameraSys, py::const_),
+                "cameraSysPrefix"_a);
+    });
+}
+
+void declareDetectorBuilder(PyDetector &parent);
+void declareDetectorPartialRebuilder(PyDetector &parent);
+void declareDetectorInCameraBuilder(PyDetector &parent);
+
+void declareDetector(lsst::utils::python::WrapperCollection &wrappers) {
+    wrappers.wrapType(PyDetector(wrappers.module, "Detector"), [](auto &mod, auto &cls) {
+        declareDetectorBuilder(cls);
+        declareDetectorPartialRebuilder(cls);
+        declareDetectorInCameraBuilder(cls);
+        cls.def("rebuild", &Detector::rebuild);
+        declare1SysMethods<CameraSys>(cls);
+        declare1SysMethods<CameraSysPrefix>(cls);
+        declare2SysMethods<CameraSys, CameraSys>(cls);
+        declare2SysMethods<CameraSys, CameraSysPrefix>(cls);
+        declare2SysMethods<CameraSysPrefix, CameraSys>(cls);
+        declare2SysMethods<CameraSysPrefix, CameraSysPrefix>(cls);
+        cls.def("getTransformMap", &Detector::getTransformMap);
+        cls.def("getAmplifiers", &Detector::getAmplifiers);
+        // __iter__ defined in pure-Python extension
+        cls.def(
+                "__getitem__",
+                [](Detector const &self, std::ptrdiff_t i) {
+                    return self[utils::python::cppIndex(self.size(), i)];
+                },
+                "i"_a);
+        cls.def("__getitem__", py::overload_cast<std::string const &>(&Detector::operator[], py::const_),
+                "name"_a);
+        cls.def("__len__", &Detector::size);
+        table::io::python::addPersistableMethods(cls);
+    });
+}
+
+void declareDetectorBuilder(PyDetector &parent) {
+    PyDetectorBuilder cls(parent, "Builder");
+    cls.def("setBBox", &Detector::Builder::setBBox);
+    cls.def("setType", &Detector::Builder::setType);
+    cls.def("setSerial", &Detector::Builder::setSerial);
+    cls.def("setPhysicalType", &Detector::Builder::setPhysicalType);
+    cls.def("setCrosstalk", &Detector::Builder::setCrosstalk);
+    cls.def("unsetCrosstalk", &Detector::Builder::unsetCrosstalk);
+    cls.def("getAmplifiers", &Detector::Builder::getAmplifiers);
+    // TODO: __iter__ defined in pure-Python extension
```

Comment:
This looks preexisting, but it looks like the "TODO" part can be removed; the `__iter__` is on `DetectorBase` in the pure-Python extension, so it should be inherited by this class.